airflow/contrib/operators/adls_to_gcs.py

```diff
@@ -0,0 +1,144 @@
+
+
+class AdlsToGoogleCloudStorageOperator(AzureDataLakeStorageListOperator):
+    """
+    Synchronizes an Azure Data Lake Storage path with a GCS bucket
+    :param path: The Azure Data Lake path to find the objects (templated)
+    :type path: str
+    :param dest_gcs: The Google Cloud Storage bucket and prefix to
+        store the objects. (templated)
+    :type dest_gcs: str
+    :param replace: If true, replaces same-named files in GCS
+    :type replace: bool
+    :param azure_data_lake_conn_id: The connection ID to use when
+        connecting to Azure Data Lake Storage.
+    :type azure_data_lake_conn_id: str
+    :param dest_google_cloud_storage_conn_id: The connection ID to use when
+        connecting to Google Cloud Storage.
+    :type dest_google_cloud_storage_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+
+    **Examples**:
+        The following Operator would copy a single file named
+        ``hello/world.avro`` from ADLS to the GCS bucket ``mybucket``. Its full
+        resulting gcs path will be ``gs://mybucket/hello/world.avro`` ::
+            copy_single_file = AdlsToGoogleCloudStorageOperator(
+                task_id='copy_single_file',
+                path='hello/world.avro',
+                dest_gcs='gs://mybucket',
+                replace=False,
+                azure_data_lake_conn_id='azure_data_lake_default',
+                google_cloud_storage_conn_id='google_cloud_default'
+            )
+
+        The following Operator would copy all parquet files from ADLS
+        to the GCS bucket ``mybucket``. ::
+            copy_all_files = AdlsToGoogleCloudStorageOperator(
+                task_id='copy_all_files',
+                path='*.parquet',
+                dest_gcs='gs://mybucket',
+                replace=False,
+                azure_data_lake_conn_id='azure_data_lake_default',
+                google_cloud_storage_conn_id='google_cloud_default'
+            )
+
+         The following Operator would copy all parquet files from ADLS
+        path ``/hello/world``to the GCS bucket ``mybucket``. ::
+            copy_world_files = AdlsToGoogleCloudStorageOperator(
+                task_id='copy_world_files',
+                path='hello/world/*.parquet',
+                dest_gcs='gs://mybucket',
+                replace=False,
+                azure_data_lake_conn_id='azure_data_lake_default',
+                google_cloud_storage_conn_id='google_cloud_default'
+            )
+    """
+    template_fields = ('path', 'dest_gcs')
+    ui_color = '#f0eee4'
+
+    @apply_defaults
+    def __init__(self,
+                 path,
+                 dest_gcs=None,
+                 replace=False,
+                 azure_data_lake_conn_id='azure_data_lake_default',
+                 google_cloud_storage_conn_id='google_cloud_default',
```

Comment:
What about leaving both `azure_data_lake_conn_id`  and `google_cloud_storage_conn_id` as Nones? 
Or maybe even making them mandatory arguments? 
Similar applies to `dest_gcs` equal to None - it is hard for me to come up with a good example when this default value makes sense.

Reply:
Agreed, I have made these changes